span.go

```diff
@@ -135,7 +137,9 @@ func (s *spanImpl) FinishWithOptions(opts opentracing.FinishOptions) {
 	s.raw.Duration = duration
 
 	s.onFinish(s.raw)
-	s.tracer.options.Recorder.RecordSpan(s.raw)
+	if s.raw.Sampled {
+		s.tracer.options.Recorder.RecordSpan(s.raw)
+	}
```

Comment:
Above is the most controversial change in this PR. Another alternative is to keep the old code, and add a `GetSampledSpans()` method to the `InMemorySpanRecorder`.